app/values/edtf_date.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+# EDTF date value object
+class EdtfDate
+  def self.for(...)
+    new(...).to_date
+  end
+
+  attr_reader :year, :month, :day, :approximate
+
+  def initialize(year:, month: nil, day: nil, approximate: false)
+    @year = year
+    @month = month
+    @day = day
+    @approximate = approximate
+  end
+
+  def to_date
+    return if year.blank?
+
+    date = Date.new(*[year, month, day].compact)
+    if month.nil?
+      date.year_precision!
+    elsif day.nil?
+      date.month_precision!
+    end
+    date.approximate! if approximate
+    date
+  end
+end
```

Comment:
Notice the value object that replaces the `EdtfSupport` "service" class does not itself define any methods to coerce to string. Why? Because Rails already has a convention for formatting `Date` objects as strings: https://api.rubyonrails.org/classes/Date.html#method-i-to_fs-label-Adding+your+own+date+formats+to+to_fs Let's build on that for our EDTF dates?